packages/foundry-sdk-generator/src/generate/betaClient/minifyBundle.ts

```diff
@@ -174,9 +278,27 @@ export class ProjectMinifier {
         this.visitNamedExports(
           namedExports,
           visitImports,
-          nodeSet,
+          nodesToKeepForThisModule,
           exportSourceFile,
         );
+
+        if (DEBUG) {
+          // eslint-disable-next-line no-console
+          console.log(
+            chalk.green(
+              `removing ${
+                chalk.white(namedExports.map(n => n.getName()).join(", "))
+              } from import set of ${logFilename(sourceFile)}`,
+            ),
+          );
+        }
+
+        for (const n of namedExports) {
+          if (visitImports.imports.has(n.getName())) {
+            nodesToKeepForThisModule.add(n);
+            visitImports.imports.delete(n.getName());
+          }
```

Comment:
Similarly, for each named export, if its in the list of things we care about we boht want to keep it in the final output and also stop looking for this identifier in future exports!